There are several warnings that result in test errors if not suppressed.

From 97ae5b72c51ba6be6cd0b3b378e0375eaeba6eec Mon Sep 17 00:00:00 2001
From: Sviatoslav Sydorenko <wk@sydorenko.org.ua>
Date: Mon, 9 Jan 2023 05:22:18 +0100
Subject: [PATCH] Suppress cgi deprecation warning in pytest

---

From 649cb51b450d6ac39c3c1b8be5dc0bb4c3e674c9 Mon Sep 17 00:00:00 2001
From: Sviatoslav Sydorenko <wk@sydorenko.org.ua>
Date: Mon, 9 Jan 2023 15:20:31 +0100
Subject: [PATCH] Suppress cheroot-caused ResourceWarning in pytest

It is related to the `webtest` helper not closing the connection
on read errors / broken connection.

--- CherryPy-18.8.0/pytest.ini
+++ CherryPy-18.8.0/pytest.ini
@@ -59,6 +59,9 @@ filterwarnings =
   ignore:the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses:DeprecationWarning
   ignore:the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses:PendingDeprecationWarning
 
+  # TODO: Replace the use of `cgi`. It seems untrivial.
+  ignore:'cgi' is deprecated and slated for removal in Python 3.13:DeprecationWarning
+
 junit_duration_report = call
 junit_family = xunit2
 junit_suite_name = cherrypy_test_suite
--- CherryPy-18.8.0/pytest.ini
+++ CherryPy-18.8.0/pytest.ini
@@ -62,6 +62,10 @@ filterwarnings =
   # TODO: Replace the use of `cgi`. It seems untrivial.
   ignore:'cgi' is deprecated and slated for removal in Python 3.13:DeprecationWarning
 
+  # TODO: Remove once `cheroot.webtest._open_url_once` is fixed to release
+  # TODO: connection properly.
+  ignore:unclosed <socket.socket fd=:ResourceWarning
+
 junit_duration_report = call
 junit_family = xunit2
 junit_suite_name = cherrypy_test_suite
